### PR TITLE
fix(indexes): allow None `primary_key` in `add_or_update` function

### PR DESCRIPTION
**Pull request by khanhnt2**
# Pull Request

## Related issue
Cannot pass `None` value to `add_or_update` function

## What does this PR do?
Replace `impl AsRef<str>` to `&str`

## PR checklist
Please check if your PR fulfills the following requirements:
- [x] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [x] Have you read the contributing guidelines?
- [x] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Review by curquiza — CHANGES_REQUESTED** [review]
Hello @khanhnt2 
thank for this PR

can you add a test passing `None` value as primary key for the `add_or_update` function?

**Comment by khanhnt2** [comment]
Hi @curquiza , sorry for the late response. I added the test case.

File: src/indexes.rs [hunk 1]
```diff
@@ -899,14 +899,12 @@ impl Index {
     pub async fn add_or_update<T: Serialize>(
         &self,
         documents: &[T],
-        primary_key: Option<impl AsRef<str>>,
+        primary_key: Option<&str>,
```
Comment (irevoire) [inline src/indexes.rs:902]: Hey, I’m not sure to understand what you changed here?

We could already send an option with the old API right?
> primary_key: Option<impl AsRef<str>>,

**Comment by curquiza** [comment]
bors try

**Review by irevoire — APPROVED** [review]
bors merge

**Review by irevoire — APPROVED** [review]
bors merge

**Review by curquiza — APPROVED** [review]
bors merge

pliiiiiiz
